exporter/tasks.py

```diff
@@ -266,59 +274,44 @@ def run(cls, filename, dry_run, **kwargs):
         )
 
         if dry_run:
-            print 'Copy S3 File: {0} to {1}'.format(
+            print('Copy S3 File: {0} to {1}'.format(
                 s3_source_filename,
-                filename)
+                filename))
         else:
             # First check to see that the export data was successfully generated
             # by looking for a marker file for that run. Return a more severe failure,
             # so that the overall environment dump fails, rather than just the particular
             # file being copied.
 
-            head_command = "aws s3api head-object --bucket {bucket} --key {key}"
-
-            marker_command = head_command.format(
-                bucket=kwargs['pipeline_bucket'],
-                key=s3_marker_filename
-            )
+            s3_client = boto3.client('s3')
 
-            source_command = head_command.format(
-                bucket=kwargs['pipeline_bucket'],
-                key=s3_source_filename
-            )
+            try:
+                log.info(f"Checking if marker file {s3_marker_filename} exsists")
+                s3_client.head_object(Bucket=kwargs['pipeline_bucket'], Key=s3_marker_filename)
+            except botocore.exceptions.ClientError as e:
+                error_code = e.response['Error']['Code']
+                if error_code == "404":
+                    error_message = f"Unable to find success marker for export {s3_marker_filename}"
+                else:
+                    error_message = f'Got {error_code} error while checking marker file'
+                raise Exception(error_message)
```

Comment:
Have you replaced `FatalTaskError` with `Exception` intentionally?

Reply:
With FatalTaskError, the script quits entirely and no output is produced. 
Whereas with Exception, a failure file is generated for the failed task with the extension '.failed'. All other tasks execute unaffected.
Refer [this method](https://github.com/open-craft/edx-analytics-exporter/blob/655d886c2c62b755379ede38e05ab7e50eff5b50/exporter/main.py#L145)